crypto/cms/cms_enc.c

```diff
@@ -74,12 +74,25 @@ BIO *cms_EncryptedContent_init_bio(CMS_EncryptedContentInfo *ec)
         if (ivlen > 0) {
             if (RAND_bytes(iv, ivlen) <= 0)
                 goto err;
-            piv = iv;
+            if (!(EVP_CIPHER_flags(ciph) & EVP_CIPH_FLAG_AEAD_CIPHER)
+                    || EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_SET_IV, ivlen,
```

Comment:
I remember that it didn't work for GCM for some reason. It's been over 5 months so need to look a bit more. I see the code changed a bit since than as well so will take a proper look if it still doesn't work even with `EVP_CTRL_SET_IVLEN`.

Reply:
This has been changed and no ctrls are needed now.